connector/signaltometricsconnector/internal/aggregator/aggregator.go

```diff
@@ -145,11 +169,27 @@ func (a *Aggregator[K]) Finalize(mds []model.MetricDef[K]) {
 				dp.Copy(a.timestamp, destCounter.DataPoints().AppendEmpty())
 			}
 		}
+		if md.Gauge == nil {
+			continue
+		}
```

Comment:
We should probably put this inside the for loop with `break`, otherwise the `delete` bit will not be invoked for anything other than gauge.